provider/provider_test.go

```diff
@@ -485,5 +484,476 @@ func TestConnectionMigration(t *testing.T) {
+                        "nonPersistentAttrs": [
+                            "ethnicity",
+                            "gender"
+                        ],
+                        "passwordComplexityOptions": null,
+                        "passwordDictionary": null,
+                        "passwordHistories": [],
+                        "passwordNoPersonalInfo": null,
+                        "passwordPolicy": "",
+                        "pingFederateBaseUrl": "",
+                        "pkceEnabled": false,
+                        "protocolBinding": "",
+                        "provider": "",
+                        "requestTemplate": "",
+                        "requiresUsername": false,
+                        "scopes": [
+                            "email",
+                            "gmail",
+                            "profile",
+                            "youtube"
+                        ],
+                        "scripts": {},
+                        "setUserRootAttributes": "on_each_login",
+                        "shouldTrustEmailVerifiedConnection": "",
+                        "signInEndpoint": "",
+                        "signOutEndpoint": "",
+                        "signSamlRequest": false,
+                        "signatureAlgorithm": "",
+                        "signingCert": "",
+                        "signingKey": null,
+                        "strategyVersion": 0,
+                        "subject": "",
+                        "syntax": "",
+                        "teamId": "",
+                        "template": "",
+                        "tenantDomain": "",
+                        "tokenEndpoint": "",
+                        "totp": null,
+                        "twilioSid": "",
+                        "twilioToken": "",
+                        "type": "",
+                        "upstreamParams": "",
+                        "useCertAuth": false,
+                        "useKerberos": false,
+                        "useWsfed": false,
+                        "userIdAttribute": "",
+                        "userinfoEndpoint": "",
+                        "validation": null,
+                        "waadCommonEndpoint": false,
+                        "waadProtocol": ""
+                    },
+                    "showAsButton": null,
+                    "realms": [
+                        "googleOauth2-b68d006"
+                    ],
+                    "strategy": "google-oauth2"
+                }
+            },
+            "metadata": {
+                "kind": "resource",
+                "mode": "client",
+                "name": "auth0"
+            }
+        }
+    ]
+`
+
+// TODO: Remove non-prc test after enabling PRC by default.
+func TestConnectionMigration(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			testutils.ReplaySequence(t, server(), prcGrpc)
+		}
+	}()
+	testutils.ReplaySequence(t, server(), nonPrcGrpc)
```

Comment:
Can we check which mode is enabled and then run the correct test? I don't love the run a test and if it panics, run another test.

The TODO needs an associated with an issue when this PR merges.
```suggestion
// TODO[pulumi/pulumi-auth0#445]: Remove non-prc test after enabling PRC by default.
func TestConnectionMigration(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			testutils.ReplaySequence(t, server(), prcGrpc)
		}
	}()
	testutils.ReplaySequence(t, server(), nonPrcGrpc)
```